File: GameServer/src/Model/Platform.php
<?php

namespace App\Model;

use App\Collision\IVisitor;

class Platform implements IGameObject, IMoveable
{
    protected $width = 175.0;
    protected $height = 45.0;

    protected $isDisappeared = false;
    protected $x;
    protected $y;

    protected $id;

    public function __construct(int $id, float $createdX, float $createdY)
    {
        $this->id = $id;

        $this->x = $createdX;
        $this->y = $createdY;
    }

    public function getIsDisappeared(): bool
    {
        return $this->isDisappeared;
    }

    public function setIsDisappeared(bool $value): void
    {
        $this->isDisappeared = $value;
    }

    public function getLeft(): float
    {
        return $this->x - $this->width / 2;
    }

    public function getRight(): float
    {
        return $this->x + $this->width / 2;
    }

    public function getTop(): float
    {
        return $this->y - $this->height / 2;
    }

    public function getBottom(): float
    {
        return $this->y + $this->height / 2;
    }

    public function accept(IVisitor $visitor): void
    {
        $visitor->visitPlatform($this);
    }

    public function getX(): float
    {
        return $this->x;
    }

    public function getY(): float
    {
        return $this->y;
    }

    public function getWidth()
    {
        return $this->width;
    }

    public function getHeight()
    {
        return $this->height;
    }

    public function setPosition(float $startX, float $startY): void
    {
        $this->x = $startX;
        $this->y = $startY;
    }

    public function updatePosition(float $elapsedTime): void
    {
    }

    public function toArray(): array
    {
        return [];
    }
}

File: MyGame/app/src/main/java/com/example/mygame/domain/platform/Platform.kt
package com.example.mygame.domain.platform

import android.graphics.Paint
import com.example.mygame.domain.IGameObject
import com.example.mygame.domain.IMoveable
import com.example.mygame.domain.IVisitor

open class Platform(createdX: Float, createdY: Float) : IMoveable, IGameObject {

    protected var paint = Paint()

    override var isDisappeared = false

    override var x = createdX
    override var y = createdY

    override val left
        get() = x - WIDTH / 2
    override val right
        get() = x + WIDTH / 2
    override val top
        get() = y - HEIGHT / 2
    override val bottom
        get() = y + HEIGHT / 2

    override fun accept(visitor: IVisitor) {
        visitor.visit(this)
    }

    override fun setPosition(startX: Float, startY: Float) {
        x = startX
        y = startY
    }

    companion object {
        const val WIDTH = 175f
        const val HEIGHT = 45f
    }
}